contracts/IdoContract/src/IdoContract.cs

```diff
@@ -456,10 +454,11 @@ private static BigInteger GetUserClaimAmountImple(byte[] key)
         public static bool ClaimToken(UInt160 user, UInt160 idoPairContractHash)
         {
             RegisteredProject project = GetRegisteredProject(idoPairContractHash);
-            if (!project.isReviewed || Ledger.CurrentIndex - project.reviewedHeight < 2 * GetVoteTimeSpan()) throw new Exception("RNE"); // project reviewed not end yet
+            ExecutionEngine.Assert(project.isReviewed && Ledger.CurrentIndex - project.reviewedHeight >= 2 * GetVoteTimeSpan(), "project review not end");
             byte[] key = GetUserClaimAmountKey(idoPairContractHash, user);
             BigInteger oldAmount = GetUserClaimAmountImple(key);
-            if (oldAmount <= 0) throw new Exception("NCT"); // no unclaimed token
+            ExecutionEngine.Assert(oldAmount > 0, "no unclaimed token");
+            SafeTransfer(project.tokenHash, Runtime.ExecutingScriptHash, user, oldAmount);
```

Comment:
Double transfer? it's already done in L464

Reply:
keep L464 and remove L461